sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesResult.cs

```diff
@@ -11,22 +12,37 @@ namespace Azure.AI.TextAnalytics
     /// </summary>
     public partial class AnalyzeHealthcareEntitiesResult : TextAnalyticsResult
     {
+        private readonly IReadOnlyCollection<HealthcareEntity> _entities;
+
         internal AnalyzeHealthcareEntitiesResult(string id, TextDocumentStatistics statistics,
-            IReadOnlyList<HealthcareEntityInternal> healthcareEntities,
-            IReadOnlyList<HealthcareRelationInternal> healthcareRelations,
-            IReadOnlyList<TextAnalyticsWarningInternal> warnings)
+            IReadOnlyList<HealthcareEntity> healthcareEntities,
+            IReadOnlyList<TextAnalyticsWarning> warnings)
             : base(id, statistics)
         {
-            Entities = Transforms.ConvertToHealthcareEntityCollection(healthcareEntities, healthcareRelations);
-            Warnings = Transforms.ConvertToWarnings(warnings);
+            _entities = healthcareEntities;
+            Warnings = warnings;
```

Comment:
for both, make sure you initialize the type and make a copy instead of just assigned.
i.e. `Warnings = new ReadOnlyCollection<TextAnalyticsWarning>(warnings);`
See the explanation on why in a comment below for the relationship dictionary

Reply:
u can also see how we are doing that with our existing GAd code https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/CategorizedEntityCollection.cs#L18